app/controllers/submissions_controller.rb

```diff
@@ -340,6 +343,20 @@ def update_files
           required_files = nil
         end
 
+        if new_url.present?
+          url_filename = params[:url_text]
+          raise I18n.t('submissions.invalid_url', item: new_url) unless is_valid_url?(new_url)
+          raise I18n.t('submissions.no_url_name', url: new_url) unless url_filename.present?
```

Comment:
I thought we discussed the URL name cannot be optional because URL names can be really long and must be sanitized first (as they will be used as the filename for the URL file).